frontend-web/webclient/app/Applications/Search.tsx

```diff
@@ -63,38 +149,58 @@ export const SearchResults: React.FunctionComponent = () => {
         );
     }, [queryParams]);
 
-    const toggleFavorite = React.useCallback(async (appName: string, appVersion: string) => {
-        await invokeCommand(UCloud.compute.apps.toggleFavorite({appName, appVersion}));
-    }, [fetch]);
+    const favoriteStatus = useSelector<ReduxObject, ApplicationSummaryWithFavorite[]>(it => it.sidebar.favorites);
+
+    const onFavorite = useCallback(async (app: ApplicationSummaryWithFavorite) => {
+        const favoriteApp = favoriteStatus.find(it => it.metadata.name === app.metadata.name);
+        const isFavorite = favoriteApp !== undefined ? true : app.favorite;
+
+        dispatch(toggleAppFavorite(app, !isFavorite));
+
+        try {
+            await callAPI(UCloud.compute.apps.toggleFavorite({
+                appName: app.metadata.name
+            }));
+        } catch (e) {
+            snackbarStore.addFailure("Failed to toggle favorite", false);
```

Comment:
Either `displayErrorMessageOrDefault(e, "Failed to toggle favorite")` or rewrite to `} catch {`